src/pages/Company/CompanyTemplate.tsx

```diff
@@ -25,82 +26,106 @@ export const CompanyTemplate: React.FC<ICompanyTemplateProps> = ({
             />
-            {StudentArray.map(student =>
+            { windowSize && windowSize.width > WindowSize.laptop ?
+              <>
+                <Route
+                  path={`${Paths.resumes}/*`}
+                  element={<HomePage students={students} studentsInBookmarks={studentsInBookmarks} />}
+                />
+                <Route
+                  path={`${Paths.bookmarks}/*`}
+                  element={<BookmarkedCVPage students={students} CVInBookmarks={studentsInBookmarks} />}
+                />
+              </> :
               <>
                 <Route
-                  path={`${Paths.resumes}/${student.id}`}
-                  element={
-                    <StudentCVPage
-                      student={student}
-                      isMarked={BookmarkedStudents.includes(student.id)}
-                      onSendInvitationClick={() => true}
-                      onAddToBookmarkClick={() => true}
-                    />}
-                  key={`${Paths.resumes}/${student.id}`}
+                  path={Paths.resumes}
+                  element={<HomePage students={students} studentsInBookmarks={studentsInBookmarks} />}
                 />
                 <Route
-                  path={`${Paths.bookmarks}/${student.id}`}
-                  element={
-                    <StudentCVPage
-                      student={student}
-                      isMarked={BookmarkedStudents.includes(student.id)}
-                      onSendInvitationClick={() => true}
-                      onAddToBookmarkClick={() => true}
-                    />}
-                  key={`${Paths.bookmarks}/${student.id}`}
+                  path={Paths.bookmarks}
+                  element={<BookmarkedCVPage students={students} CVInBookmarks={studentsInBookmarks} />}
                 />
+                {students.map(student =>
+                  <>
+                    <Route
+                      path={`${Paths.resumes}/${student.id}`}
+                      element={
+                        <StudentCVPage
+                          student={student}
+                          isMarked={studentsInBookmarks.includes(student.id)}
+                          onSendInvitationClick={() => true}
+                          onAddToBookmarkClick={() => true}
+                        />}
+                      key={`${Paths.resumes}/${student.id}`}
+                    />
+                    <Route
+                      path={`${Paths.bookmarks}/${student.id}`}
+                      element={
+                        <StudentCVPage
+                          student={student}
+                          isMarked={studentsInBookmarks.includes(student.id)}
+                          onSendInvitationClick={() => true}
+                          onAddToBookmarkClick={() => true}
+                        />}
+                      key={`${Paths.bookmarks}/${student.id}`}
+                    />
```

Comment:
Almost the same.

Reply:
As in your suggestion I could use just `element={StudentCVPage}` , but I don't have any state container yet. So I have to request student info here and pass it to other pages as prop. I think this way is better than make requests on each page.